cep-XXXX.md

```diff
@@ -0,0 +1,230 @@
+
+- `./Library/`, uses the same directories as the Unix structure listed above. Most packages will populate this directory.
+- Python interpreters and Python-related packages stay in the root-level:
+  - `./DLLs/`: Compiled Python extensions (`.pyd`)
+  - `./include/`: Python development headers (`.h`).
+  - `./Lib/`: Equivalent to `./lib/pythonX.Y/site-packages` on Unix.
+  - `./libs/`: Python `.lib` files.
+  - `./Scripts/`: Python entry points (`.exe` trampolines and the corresponding `-script.py` files).
+  - `./Tools/`: Miscellaneous Python scripts.
+  - `./python*.(exe|dll|pdb)`: The Python interpreter executables and libraries.
+- Other interpreted languages might also populate the root-level directly, specially if they rely on `noarch: generic` packages. Some examples:
+  - Node.js places its executables in the root level, and leaves everything else under `./node_modules/`.
+  - R installs to `./lib/R/` using a Unix-style directory structure, but also places some executables in `./Scripts/`.
+  - Ruby installs directly to the root level, using a Unix-style directory structure.
+
+Regardless the platform, within these directories there are some special paths that conda clients SHOULD handle in a specific way:
+
+- `./etc/conda/*.d` directories
+- `./(bin|Scripts)/.{package-name}-{action}.{extension}` scripts
+- `./condarc` configuration files
+
+#### `etc/conda/*.d` directories
+
+This special directory stores configuration files that can modify the behavior of the conda client at runtime. Packages are allowed to ship files under this directory.
+
+The following files and directories MUST be handled by the conda client:
+
+- `./env_vars.d/`: Directory containing JSON documents, each providing a dictionary that maps strings to strings, which are understood as environment variables. The JSON documents SHOULD be loaded alphabetically, with later documents overriding earlier ones. These environment variables SHOULD be exported on environment activation. When both `env_vars.d/` and `conda-meta/state` are present, the latter is loaded last and can override the previously defined variables.
+- `./activate.d/`: Directory containing shell scripts. They SHOULD be executed on environment activation, in alphabetical order.
+- `./deactivate.d/`: Directory containing shell scripts. They SHOULD be executed on environment deactivation, in reverse alphabetical order.
+
+### Pre- and post-link/unlink scripts
+
+The `bin/` and `Scripts/` directories usually contain executables populated by the packages themselves. There are four special paths that SHOULD be handled by conda clients. For files named like `.{package-name}-{action}.{extension}`, where `{package-name}` corresponds to the package name and `{extension}` is either `sh` (Unix) or `bat` (Windows), there are four possible `{action}` values and associated behaviors:
+
+- `pre-link`: Executed before a package is installed / linked. Deprecated; conda clients SHOULD ignore them.
```

Comment:
conda currently runs `pre-link` scripts. 

IMHO all of these scripts should be avoided but there are valid uses cases for all of them that do not yet have replacements. https://docs.conda.io/projects/conda-build/en/stable/resources/link-scripts.html touches on this.